pkg/datasource.go

```diff
@@ -153,29 +160,81 @@ func (o *OCIDatasource) QueryData(ctx context.Context, req *backend.QueryDataReq
 }
 
 func (o *OCIDatasource) testResponse(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
-	return &backend.QueryDataResponse{}, nil
-	// var ts GrafanaCommonRequest
-
-	// query := req.Queries[0]
-	// if err := json.Unmarshal(query.JSON, &ts); err != nil {
-	// 	return &backend.QueryDataResponse{}, err
-	// }
-
-	// //o.logger.Error("ts.Com is " + ts.Compartment)
-	// listMetrics := monitoring.ListMetricsRequest{
-	// 	CompartmentId: common.String(ts.Compartment),
-	// }
-	// reg := common.StringToRegion(ts.Region)
-	// o.metricsClient.SetRegion(string(reg))
-	// res, err := o.metricsClient.ListMetrics(ctx, listMetrics)
-	// if err != nil {
-	// 	return &backend.QueryDataResponse{}, err
-	// }
-	// status := res.RawResponse.StatusCode
-	// if status >= 200 && status < 300 {
-	// 	return &backend.QueryDataResponse{}, nil
-	// }
-	// return nil, errors.Wrap(err, fmt.Sprintf("list metrircs failed %s %d", spew.Sdump(res), status))
+	var ts GrafanaCommonRequest
+	query0 := req.Queries[0]
+
+	if err := json.Unmarshal(query0.JSON, &ts); err != nil {
+		return &backend.QueryDataResponse{}, err
+	}
+	o.logger.Debug("ts.TenancyOCID:", "*ts.TenancyOCID", ts.TenancyOCID)
+
+	compreq := identity.ListCompartmentsRequest{AccessLevel: identity.ListCompartmentsAccessLevelAny,
+		CompartmentIdInSubtree: common.Bool(true),
+		Limit:                  common.Int(185),
+		CompartmentId:          common.String(ts.TenancyOCID)}
+
+	ListCompartments, err := o.identityClient.ListCompartments(ctx, compreq)
+	o.logger.Debug("ListCompartments:", "*ListCompartments", ListCompartments.Items)
+
+	if err == nil {
+		for _, compartitem := range ListCompartments.Items {
+			listLogsGroup := logging.ListLogGroupsRequest{
+				CompartmentId: common.String(*compartitem.Id),
+			}
+			o.logger.Debug("*compartitem.Id :", "*compartitem.Id", *compartitem.Id)
+			listLogsGroups, err := o.loggingClient.ListLogGroups(ctx, listLogsGroup)
```

Comment:
While this is a very creative solution, however I have several issues with it:

1. Performing these operations across all compartments in a tenancy (which is the only way the test would be meaningful) seems very excessive and could take quite a long time in a large tenancy such as CN Devcorp.
2. More importantly the principals used by Grafana may not be permitted to list log groups, list logs, and/or search logs in all (or any) of the tested compartments. This will lead to 404 errors ("NotAuthorizedOrNotFound. Authorization failed or requested resource not found") being returned for these operations which admittedly won't cause an issue for the plugin. But this is very poor behavior from the plugin as it could cause 100s of 404 errors to be returned which could be seen by security personnel as some type of probing attack, especially in a production environment.

This is why I believe we still need to have the user provide a "default compartment" as part of the datasource configuration, similar to the default region the user needs to provide currently. Otherwise the plugin has to perform a very large number of operations to perform the test, all of which could fail depending on the principals used by Grafana.